src/main.rs

```diff
@@ -78,13 +76,17 @@ pub struct OsInputOutput {
     pub network_frames: Box<dyn DataLinkReceiver>,
     pub get_open_sockets: fn() -> HashMap<Connection, String>,
     pub keyboard_events: Box<dyn Iterator<Item = Event> + Send>,
-    pub lookup_addr: Box<dyn Fn(&IpAddr) -> Option<String> + Send>,
+    pub resolver_factory: Box<dyn FnOnce(IpTable) -> Result<Box<dyn Lookup>, failure::Error>>,
```

Comment:
clippy complains about this type and the one below being to complex, and that we should create an alias using `type`. I think that is indeed a good idea. I've notice other clippy warnings in the project, if you agree I could fix it either in this PR or in a dedicated issue.

Reply:
That would be great! It would be easier for me if it's done in a separate PR, yeah.